packages/flutter/lib/src/painting/_image_provider_web.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' show hashValues;
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+import 'image_provider.dart' as image_provider;
+import 'image_stream.dart';
+
+/// The dart:html implemenation of [image_provider.NetworkImage].
+class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkImage> implements image_provider.NetworkImage {
+  /// Creates an object that fetches the image at the given URL.
+  ///
+  /// The arguments must not be null.
+  const NetworkImage(this.url, {this.scale = 1.0, this.headers})
+      : assert(url != null),
+        assert(scale != null);
+
+  @override
+  final String url;
+
+  @override
+  final double scale;
+
+  @override
+  final Map<String, String> headers;
+
+  @override
+  Future<NetworkImage> obtainKey(image_provider.ImageConfiguration configuration) {
+    return SynchronousFuture<NetworkImage>(this);
+  }
+
+  @override
+  ImageStreamCompleter load(image_provider.NetworkImage key) {
+    return MultiFrameImageStreamCompleter(
+      codec: _loadAsync(key),
+      scale: key.scale,
+      informationCollector: () sync* {
+        yield DiagnosticsProperty<image_provider.ImageProvider>('Image provider', this);
+        yield DiagnosticsProperty<NetworkImage>('Image key', key);
+      },
+    );
+  }
+
+  Future<ui.Codec> _loadAsync(NetworkImage key) async {
```

Comment:
Do we need to do something here to get the new `ImageChunkEvent` feature working on web? It seems like the APIs for IO and Web won't match for `NetworkImage` otherwise

Reply:
is it possible to get the chunk events on the web?

Reply:
It doesn't look like you can get chunk events for the images, just an all-or-nothing event for when the image completely finishes loading.